inc/umock_c/umocktypes_struct.h

```diff
@@ -43,11 +43,11 @@ extern "C" {
 
 /*Codes_SRS_UMOCKTYPES_STRUCT_42_007: [ umocktypes_are_equal_<type> shall call umocktypes_are_equal for each field in type. ]*/
 #define UMOCK_STRUCT_ARE_EQUAL_STRUCT_FIELD(count, field_type, field_name) \
-    umocktypes_are_equal(MU_TOSTRING(field_type), &left->field_name, &right->field_name) &
+    umocktypes_are_equal(MU_TOSTRING(field_type), (const void*)&left->field_name, (const void*)&right->field_name) &

```

Comment:
>umocktypes_are_equal [](http://example.com/codeflow?start=4&length=20)

What happens if umocktypes_are_equal would take the pointer to const volatile instread of just pointer to const?

Wouldn't that save some casts here and there?

Reply:
If we changed that signature from:

```
int umocktypes_are_equal(const char* type, const void* left, const void* right)
```

to

```
int umocktypes_are_equal(const char* type, const void* volatile left, const void* volatile right)
```

Then it still complains. Likely because there is some double indirection (we pass the address-of the field, so really it's a `const int* volatile*` (I don't know where the modifiers and stars even go when you start to write out pointers to pointers with modifiers)).

So I still get 

> warning C4090: 'function': different 'volatile' qualifiers

I think the cast fix is simpler